src/engine.rs

```diff
@@ -1486,6 +1614,26 @@ impl CaCert {
     pub fn rpki_notify(&self) -> Option<&uri::Https> {
         self.cert.rpki_notify()
     }
+
+    /// Returns whether the CA is in a different repository from its parent.
+    ///
+    /// This is just a quick check and may report a switch when in fact there
+    /// isn’t one.
+    fn repository_switch(&self) -> bool {
+        let parent = match self.parent.as_ref() {
+            Some(parent) => parent,
+            None => return true,
+        };
+
+        match self.rpki_notify() {
+            Some(rpki_notify) => {
+                Some(rpki_notify) != parent.rpki_notify()
+            }
+            None => {
+                self.ca_repository.module() != parent.ca_repository.module()
```

Comment:
Sounds like this is comparing the directories on the (possibly same) rsync server, ergo comparing 'publication points' instead of 'repositories'.
Also, is there a reason not to use .canonical_module() here?

Reply:
The module returned by uri::Rsync::module is the URI up to and including the first path component. Which is what the metrics are relative for.

It is not using canonical module because that is more expensive. Since accidentally claiming a switch doesn’t break anything and differences in case in the host name (which is what canonical module checks for) are rare, this seemed a good trade-off.